src/explorer/mod.rs

```diff
@@ -304,26 +330,26 @@ impl<'a> GitExplorer {
         }
 
         let stop_condition = self.stop_conditions.get(self.stop_condition_i).unwrap();
-        let abort_next = match stop_condition {
+        self.abort = match stop_condition {
```

Comment:
This line will need to mutate `GitExplorer` instance, and this cascades the error until reaching this [point](https://github.com/KarlHeitmann/git_explorer/pull/3/files#diff-94650c3b1172d7a5565e9239f9dd69f75e342eed9dd53ed6ed8f00321b1e76f2R175)